lib/spack/spack/spec.py

```diff
@@ -2385,6 +2385,7 @@ def to_node_dict(self, hash=ht.dag_hash):
                     "parameters": {
                         "deptypes": dt.flag_to_tuple(dspec.depflag),
                         "virtuals": dspec.virtuals,
+                        "direct": dspec.direct,
```

Comment:
Can we avoid adding `"direct"` if the spec is concrete? This attribute has some meaning only for abstract specs, so I would keep the concrete specs consistent with the current format. Besides that, currently this PR shows this:
<p align="center">
<img src="https://github.com/user-attachments/assets/8df67c20-6735-435c-ab79-3bc25e4a84fb" alt="json dep" width="60%" height="auto">
</p>

for concrete specs :grimacing:

Reply:
I think I should probably fix the fact that `edge.direct` returns `False` for direct edges on concrete specs.

Reply:
Fine if we want to fix the `direct` attribute for the spec object (or document it has no meaning on them), but I think we shouldn't add more entries to concrete specs JSON. For what is worth, this attribute is just meant to discriminate `%` from `^` on abstract specs.